Bio/SeqRecord.py

```diff
@@ -1409,12 +1409,17 @@ def key_fun(f):
         elif annotations:
             # Copy the old annotations,
             answer.annotations = self.annotations.copy()
-        if isinstance(letter_annotations, dict):
-            answer.letter_annotations = letter_annotations
-        elif letter_annotations:
-            # Copy the old per letter annotations, reversing them
-            for key, value in self.letter_annotations.items():
-                answer.letter_annotations[key] = value[::-1]
+
+        # _per_letter_annotations is lazily initialized when
+        # letter_annotations is accessed, so do nothing if uninitialized.
+        # See issue #5015
```

Comment:
I would drop this comment line - we have git blame, and the comment and test alone are clear.